### PR TITLE
[Changelog] add filter bar to changelog page

### PR DESCRIPTION
**Pull request by ryancurtis1**
Adds filter options between Announcements (as designated in an individual post) and updates (anything else)

https://www.loom.com/share/a793b898a05d41d8a934a63939dec78d?sid=b9d3f823-4bbf-41e2-b470-be1509a643c8

### CODE REVIEW

File: components/changelog-index.js [hunk 1]
```diff
@@ -56,17 +56,46 @@ export default function ChangelogIndex({ more = "Read more" }) {
...
   useEffect(() => {
-    const morePages = allPages.slice(pageIndex, pageIndex + itemsPerPage);
-    setDisplayedPages((prev) => [...prev, ...morePages]);
-  }, [pageIndex]);
+    const morePages = allPages.filter(page => {
+      switch (filter) {
+        case 'updates':
+          return page.frontMatter?.isAnnouncement !== true;
+        case 'announcements':
+          return page.frontMatter?.isAnnouncement === true;
```
Comment (tiffanyqi) [inline components/changelog-index.js]: do we need the conditional part, could we just `page.frontMatter?.isAnnouncement` or `!page.frontMatter?.isAnnouncement`

File: components/changelog-index.js [hunk 2]
```diff
@@ -58,55 +56,95 @@ export default function ChangelogIndex({ more = "Read more" }) {
...
   // Load initial or additional pages
   useEffect(() => {
-    const morePages = allPages.slice(pageIndex, pageIndex + itemsPerPage);
-    setDisplayedPages((prev) => [...prev, ...morePages]);
-  }, [pageIndex]);
+    const morePages = allPages.filter(page => {
+      switch (filter) {
+        case 'updates':
+          return page.frontMatter?.isAnnouncement !== true;
+        case 'announcements':
```
Comment (tiffanyqi) [inline components/changelog-index.js]: i'd use an enum here

File: components/changelog-index.js [hunk 3]
```diff
@@ -58,55 +56,95 @@ export default function ChangelogIndex({ more = "Read more" }) {
...
+    }).slice(0, pageIndex + itemsPerPage);
+    setDisplayedPages(morePages);
+  }, [pageIndex, filter]);
 
   const loadMore = () => {
     setPageIndex((prev) => prev + itemsPerPage);
   };
 
+  const filterButton = (id, label) => {
+    let className = 'changelogFilterButton';
```
Comment (tiffanyqi) [inline components/changelog-index.js]: same here...? assuming it's used elsewhere

File: components/changelog-index.js [hunk 4]
```diff
@@ -36,7 +35,6 @@ const renderMedia = (page) => {
       <iframe
```
Comment (tiffanyqi) [inline components/changelog-index.js]: nit can we actually turn this into a tsx file...?

**Review by tiffanyqi — APPROVED** [review]
approving with a few change requests

File: components/changelog-index.js (hunk 4 above)
Comment (ryancurtis1) [reply components/changelog-index.js]: 👍

**Pull request merged**